### PR TITLE
Enable torchx build on Windows

### PR DESCRIPTION
**Pull request by howard0su**
While I am here, change the URL pattern with version and target. At least this pattern works for 1.9.0, 1.9.1 and 1.10.0

### CODE REVIEW

**Review by josevalim — APPROVED** [review]

File: torchx/mix.exs [hunk 1]
```diff
@@ -92,20 +93,24 @@ defmodule Torchx.MixProject do
...
             # MacOS
-            "https://download.pytorch.org/libtorch/cpu/libtorch-macos-1.9.1.zip"
+            "https://download.pytorch.org/libtorch/#{@libtorch_target}/libtorch-macos-#{@libtorch_version}.zip"
 
-          os ->
+          {:win32, :nt} ->
+            #Windows
+            "https://download.pytorch.org/libtorch/#{@libtorch_target}/libtorch-win-shared-with-deps-#{@libtorch_version}%2B#{@libtorch_target}.zip"
+
+            os ->
```
Comment (josevalim) [inline torchx/mix.exs:113]:
```suggestion
          os ->
```

**Comment by josevalim** [comment]
We need to list the allowed actions, so I did so and restarted CI. :)

**Comment by josevalim** [comment]
@howard0su CI ran now but it seems some headers are missing. :)

**Comment by howard0su** [comment]
> @howard0su CI ran now but it seems some headers are missing. :)

Because of the cache key of libtorch doesn't contain os name. I fixed that. Now it fails with 3 doctests now.

File: torchx/Makefile.win [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	@ if not exist "$(CMAKE_BUILD_DIR)" mkdir "$(CMAKE_BUILD_DIR)"
+	@ if not exist "$(PRIV_DIR)/$(LIBTORCH_BASE)" mkdir "$(PRIV_DIR)/$(LIBTORCH_BASE)"
+	@ xcopy /Y /Q "$(LIBTORCH_DIR)/lib"\*.dll "$(PRIV_DIR)/$(LIBTORCH_BASE)"
+	@ cd "$(CMAKE_BUILD_DIR)" &&\
+		cmake -DCMAKE_PREFIX_PATH="$(LIBTORCH_DIR)" -DC_SRC="$(C_SRC)" \
+				-DLIBTORCH_BASE="$(LIBTORCH_BASE)" -DERTS_INCLUDE_DIR="$(ERTS_INCLUDE_DIR)" \
+				-S "$(MAKEDIR)" $(CMAKE_CONFIGURE_FLAGS) && \
+		cmake --build . $(CMAKE_BUILD_FLAGS) --config Release
+	@ xcopy /Y /Q "$(CMAKE_BUILD_DIR)/Release"\*.dll "$(PRIV_DIR)"
+	@ echo Please Add "$(PRIV_DIR)/$(LIBTORCH_BASE)" to your PATH env var.
```
Comment (josevalim) [inline torchx/Makefile.win:17]:
```suggestion
	@ echo Please add "$(PRIV_DIR)/$(LIBTORCH_BASE)" to your PATH env var.
```

**Review by josevalim — APPROVED** [review]

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**